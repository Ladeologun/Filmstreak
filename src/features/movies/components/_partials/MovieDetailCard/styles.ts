import { StyleSheet } from "react-native";
import { COLORS, Fonts } from "~styles";

const styles = StyleSheet.create({
    
    votesandpopularity:{
        flexDirection:"row",
        marginVertical:16
    },
    categorycontainer:{
        flexDirection:"row",
        alignItems:"center",
        justifyContent:'center',
        paddingHorizontal:8,
        paddingVertical:3,
        borderRadius:8,
        backgroundColor:COLORS.LIGHT_GRAY,
    },
    categorttext:{
        color:"#fff",
        fontSize:14, 
        lineHeight: 22,
       fontFamily: Fonts.RalewayBold
    },
    title:{
        color: COLORS.WHITE,
        fontSize:24,
        fontFamily: Fonts.CreepsterRegular
    },
    imageStyle:{
        width: 15, 
        height: 15, 
        marginRight: 5
    },
    btnStyle: {
        alignSelf: 'flex-start', 
        marginTop: 0, 
        paddingVertical: 8
    }

})

export default styles;