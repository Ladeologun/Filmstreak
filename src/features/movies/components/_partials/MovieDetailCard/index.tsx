import React from 'react';
import {Image, Text, View} from 'react-native';
import styles from './styles';
import {STAR_LOGO} from '~movies/constants';
import WatchlistButton from '../WatchlistButton/Index';
import {LimitTextComponent} from '~components';

interface Iprops {
   id: number;
   title: string;
   language: string;
   popularity: number;
   rating: number;
   addTowishlist: (status: boolean) => void;
   isAddingMovieToWishlist: boolean;
   isMovieOnWishlist: boolean;
   isCheckingMovieState: boolean;
}

const MovieDetailCard: React.FC<Iprops> = ({
   title,
   popularity,
   rating,
   addTowishlist,
   isAddingMovieToWishlist,
   isMovieOnWishlist,
   isCheckingMovieState,
}) => {
   return (
      <React.Fragment>
         <LimitTextComponent text={title} limit={25} textStyle={styles.title} />
         <View style={styles.votesandpopularity}>
            <View style={[styles.categorycontainer]}>
               <Text style={styles.categorttext}>{popularity}</Text>
            </View>
            <View style={[styles.categorycontainer, {marginLeft: 8}]}>
               <Image source={STAR_LOGO} resizeMode="contain" style={styles.imageStyle} />
               <Text style={styles.categorttext}>{rating}</Text>
            </View>
         </View>

         {!isCheckingMovieState && (
            <WatchlistButton
               isOnwishlist={isMovieOnWishlist}
               btnStyle={styles.btnStyle}
               addingTowishlist={isAddingMovieToWishlist}
               handleBtnPress={() => {
                  if (isMovieOnWishlist) {
                     addTowishlist(false);
                  } else {
                     addTowishlist(true);
                  }
               }}
            />
         )}
      </React.Fragment>
   );
};

export default React.memo(MovieDetailCard);
