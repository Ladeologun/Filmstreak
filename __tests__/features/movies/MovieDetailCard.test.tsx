import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import MovieDetailCard from '~movies/components/_partials/MovieDetailCard';


test('renders correctly', () => {
    const tree = renderer.create(
        <MovieDetailCard
            language='en' 
            id={2}
            title={"new dragon show"}
            popularity={7}
            rating={7.6}
            addTowishlist={()=>{}}
            isAddingMovieToWishlist={false}
            isMovieOnWishlist={true}
            isCheckingMovieState={false}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});
