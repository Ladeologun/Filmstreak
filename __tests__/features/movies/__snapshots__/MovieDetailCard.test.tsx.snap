// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <Text
    style={
      {
        "color": "#ffffff",
        "fontFamily": "Creepster-Regular",
        "fontSize": 24,
      }
    }
  >
    new dragon show
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#363636",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "paddingVertical": 3,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Raleway-Bold",
            "fontSize": 14,
            "lineHeight": 22,
          }
        }
      >
        7
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#363636",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "paddingVertical": 3,
          },
          {
            "marginLeft": 8,
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={
          {
            "testUri": "../../../src/assets/drawables/star.png",
          }
        }
        style={
          {
            "height": 15,
            "marginRight": 5,
            "width": 15,
          }
        }
      />
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Raleway-Bold",
            "fontSize": 14,
            "lineHeight": 22,
          }
        }
      >
        7.6
      </Text>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignSelf": "center",
            "backgroundColor": "#363636",
            "borderRadius": 5,
            "flexDirection": "row",
            "marginTop": 16,
            "paddingHorizontal": 8,
            "paddingVertical": 5,
          },
          {
            "backgroundColor": "red",
          },
          {
            "alignSelf": "flex-start",
            "marginTop": 0,
            "paddingVertical": 8,
          },
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/drawables/love.png",
          }
        }
        style={
          {
            "height": 20,
            "marginRight": 10,
            "tintColor": "#ffffff",
            "width": 20,
          }
        }
      />
      <Text
        style={
          {
            "color": "#ffffff",
            "fontFamily": "Raleway-Bold",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Saved to wishlist
      </Text>
    </View>
  </View>,
]
`;
